### PR TITLE
Dmac

### PR DESCRIPTION
**Pull request by xd009642**
Started work at the register definitions for the DMAC

### CODE REVIEW

**Comment by xd009642** [comment]
I was wondering if it could be a good idea to split the DMAC registers into two PRs, this one for the common registers and then another for the channel registers. Naturally up to you, I'm still keeping each register to it's own commit to make reviewing easier :smile:

File: ps2.xml [hunk 1]
```diff
@@ -2457,6 +2457,13 @@
                     <name>D_STADR</name>
                     <description>DMAC stall address</description>
                     <addressOffset>0x6060</addressOffset>
+                    <fields>
+                        <field>
+                            <name>ADDR</name>
+                            <description>Stall address (qword alignment)</description>
+                            <bitRange>[30:0]</bitRange>
```
Comment (ZirconiumX) [inline ps2.xml:2464]: If this field is qword-aligned...wouldn't it make sense to truncate the bottom bits, too?

**Comment by xd009642** [comment]
I also haven't set any reset values and wondering if I should be for this one given it's mostly r/w.

File: ps2.xml [hunk 2]
```diff
@@ -2433,11 +2433,25 @@
...
                 </register>
                 <register>
                     <name>D_RBOR</name>
                     <description>DMAC ring buffer offset</description>
                     <addressOffset>0x6050</addressOffset>
+                    <fields>
+                        <field>
+                            <name>ADDR</name>
+                            <description>Ring buffer offset address (qword alignment)</description>
+                            <bitRange>[30:0]</bitRange>
```
Comment (ZirconiumX) [inline ps2.xml:2452]: Same thing as with `D_STADR`

File: ps2.xml (hunk 1 above)
Comment (xd009642) [reply ps2.xml:2464]: I was just typing something about that, I was going to say they're always meant to be zero so I could truncate them but then as they're always addresses maybe the numbers should be kept to the same magnitude

**Pull request merged**